server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -139,6 +139,7 @@ class ObjectService(
         updateMetadataResponse =>
           updateMetadataResponse match {
             case UpdateMetadataResponse.DirectMetadataUpdate => IO.unit
+              metadataCacheAlg.updateLock(localPath,lock)
```

Comment:
this is the bug fix. Previously, when lock is updated properly in gcp, local cache for the lock is not updated properly